From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 20 Aug 2020 19:24:13 -0700
Subject: [PATCH] Fix MC-99259 Wither Boss Bar doesn't update until
1.17 Update: This issue is marked as fixed on 1.17 - yeet!
 invulnerability period is over

Resolved https://bugs.mojang.com/browse/MC-99259

diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index edd231568b75330d0cffbecb03a7e9dbc55d5f94..1f330d852eb9b3a36570542e10a88ae065798714 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -391,8 +391,9 @@ public class WitherBoss extends Monster implements RangedAttackMob {
                 this.heal(1.0F, EntityRegainHealthEvent.RegainReason.REGEN); // CraftBukkit
             }
 
-            this.bossEvent.setPercent(this.getHealth() / this.getMaxHealth());
+            //this.bossBattle.setProgress(this.getHealth() / this.getMaxHealth()); // Paper - Moved down
         }
+        this.bossEvent.setPercent(this.getHealth() / this.getMaxHealth()); // Paper - Fix MC-99259 (Boss bar does not update until Wither invulnerability period ends)
     }
 
     public static boolean canDestroy(BlockState block) {
